### PR TITLE
Stabilize num::NonZeroU*

### PR DESCRIPTION
**Pull request by SimonSapin**
Tracking issue: https://github.com/rust-lang/rust/issues/49137

### CODE REVIEW

**Comment by alexcrichton** [comment]
@bors: r+

🎉

File: src/test/rustdoc/synthetic_auto/no-redundancy.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ where
 
 // @has no_redundancy/struct.Outer.html
 // @has - '//*[@id="synthetic-implementations-list"]/*[@class="impl"]//*/code' "impl<T> Send for \
-// Outer<T> where T: Copy + Send"
+// Outer<T> where T: Send + Copy"
```
Comment (kennytm) [inline src/test/rustdoc/synthetic_auto/no-redundancy.rs]: Huh, this is strange. cc @GuillaumeGomez

Comment (SimonSapin) [reply src/test/rustdoc/synthetic_auto/no-redundancy.rs]: Oh, this is a rebase of an old branch, this change is probably not necessary anymore. Removed.

**Pull request merged**